download-maven-plugin/src/main/java/com/googlecode/download/maven/plugin/internal/cache/Index.java

```diff
@@ -0,0 +1,32 @@
+package com.googlecode.download.maven.plugin.internal.cache;
+
+/**
+ * Cache index.
+ * @author Paul Polishchuk
+ * @since 1.3.1
+ */
+interface Index {
+
+    /**
+     * Adds given path to the index using url parameter as a key.
+     * @param url index key
+     * @param path index value
+     */
+    void put(String url, String path);
```

Comment:
@mickaelistria OK, will elaborate :)
1) I don't like wrappers, which do nothing apart wrapping. In our case, wrapper provides additional functionality, such as persistence :)
2) Basically, If the user will pass URL instead of a simple string I agree to store URL instead of String as a key.
But, I would like to make this change as a separate PR, if you don't mind
3) "whether readers/implementors/parser/static analyzers.. will more easily understand what's expected".  According to this point we should forget about usage of common java types and wrap them all by custome types just to make them explicit for readers ... This could be beneficial in **some** cases (like when adding new useful functionality, validation etc.) but it should not be a goal by itself!

Summary:
- I agree to pass URL instead of a sting (will be done as a separate PR, as  many other places should be fixed in case of a change)
- still insist on String path instead of a File :)